src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -64,6 +104,35 @@ public static float cosinesimilOptimized(float[] queryVector, float[] inputVecto
         return (float) (dotProduct / (Math.sqrt(normalizedProduct)));
     }
 
+    /**
+     * Whitelisted method for users that can be used script to avoid repeated calculation of normalization
+     * for query vector for each filtered documents
+     * example
+     *  "script": {
+     *         "source": "cosineSimilarity(params.query_vector, docs[field], 1.0) ",
```

Comment:
This should be cosineSimilarityOptimized, no?

Reply:
Ack.

Reply:
addressed your rename comment, hence keeping cosineSimilarity as overloaded.